doc/languages.rst

```diff
@@ -949,6 +951,19 @@ to view the docstring of the syntax checker.  Likewise, you may use
          A list of additional library directories. Relative paths are relative
          to the buffer being checked.
 
+   .. syntax-checker:: rust-cargo-clippy
```

Comment:
Agreed on `flycheck-rust-library-path`.  I had plan to phase it out of `rust-cargo`, so no need to rely on it.

As for the others, the reasoning is sound.  Thank you @dzamlo for taking the time to outline it.